### ISSUE TITLE
Questions re. 1)IDR and 2)broad peaks

### ISSUE THREAD

**Issue by PocketChangeJ** [ENCODE-DCC/chip-seq-pipeline2#30]
Hi guys,
It's more of a question - not an issue, hope that's ok:
I'm using your pipeline to look at some of the published ChIP results, and most are unreplicated. From what I noticed and understand is that the pipeline will in most cases still run the self-consistency IDR with pseudoreplicates and so on, but it doesn't always do that and sometimes returns just overlap (?) between replicate and pseudoreplicate without attempting IDR? Would it be possible for you to explain when one vs the other happens?
And second thing - what would you recommend in case of ChIPs that give 'broad' signals, like h3k27me3 seems to, or mixed like Pol2 apparently does? Is there a way to adjust the peak calling parameters etc?

**Comment by leepc12** [comment]
1) self-consistency IDR
Generation of IDR peaks has nothing to do with the number of replicates. It depends on the pipeline type (`histone` or `tf`). You will get IDR peaks for `tf` only.

2) broad peaks
We disabled gapped/broad peak generation because MACS2 has some flaws in its algorithm for them. But if you still want to get possibly flawed gapped/broad peaks then use our old pipeline and modify https://github.com/kundajelab/chipseq_pipeline/blob/master/chipseq.bds#L34 to enable it.

**Comment by PocketChangeJ** [comment]
I see, that explains it.
Thank you!

**Comment by icanwinwyz** [comment]
> 1. self-consistency IDR
>    Generation of IDR peaks has nothing to do with the number of replicates. It depends on the pipeline type (`histone` or `tf`). You will get IDR peaks for `tf` only.
> 2. broad peaks
>    We disabled gapped/broad peak generation because MACS2 has some flaws in its algorithm for them. But if you still want to get possibly flawed gapped/broad peaks then use our old pipeline and modify https://github.com/kundajelab/chipseq_pipeline/blob/master/chipseq.bds#L34 to enable it.

Hi leepc12, 
I just deployed your ChIP-seq pipeline on our HPC and successfully finished a test run. Thanks for making this great pipeline!

My questions are related to this topic so I won't open a new ticket:

1.  Would you consider integrating IDR analysis for "histone"?

2. I searched online and failed to find any information talking about the flaws in MACS2's algorithm for broad peak calling. Could you explain a little bit or show me where I can find the information about this? Our project is heavily relying on the broad peak calling by using MACS2. If it turns out the algorithm is wrong, we need to redo the analysis. 

Many thanks in advance!!

**Comment by akundaje** [comment]
>
>
>    1.
>
>
>    Would you consider integrating IDR analysis for "histone"?
>
>
I dont recommend using IDR for histone marks. This is because unlike TFs,
histones have a much smoother and more variable dynamic range of signal and
learning a cutoff based on rank consistency especially with MACS2 scores
makes it too conservative. The naive overlap strategy we have implemented
typically works very well.

>
>    1.
>    2.
>
>    I searched online and failed to find any information talking about the
>    flaws in MACS2's algorithm for broad peak calling. Could you explain a
>    little bit or show me where I can find the information about this? Our
>    project is heavily relying on the broad peak calling by using MACS2. If it
>    turns out the algorithm is wrong, we need to redo the analysis.
>
>
MACS2's broad peak calling routines have many serious issues.

The scoring method for broad peaks (foldchange and pvalue) is quite
arbitrary in the way it combines scores of individual peaks that make up a
broadpeak it really does not rank peaks in any intuitive way. Hence if you
threshold the peaks using these scores you get very strange
inconsistencies. Eg. several strong narrowpeaks are often missed by
broadpeaks for the same thresholds. The peaks that have higher scores than
others often do not make any sense when you actually inspect them.
Reproducibility across replicates is extremely difficult to estimate
because the peak boundaries themselves are unstable and the ranking
measures are unstable.

So to summarize the broadPeaks that are called are MACS2 are often
reasonable in terms of their locations. But the ranking of peaks are often
totally off. So there is a lot that it can miss and reproducibility and
stability of the peaks and their boundaries is poor.

-Anshul.
